src/apps/chifra/cmd/pinman/handle_list.go

```diff
@@ -32,29 +34,43 @@ func HandleList(format string) {
 	sort.Slice(manifestData.NewPins, func(i, j int) bool {
 		iPin := manifestData.NewPins[i]
 		jPin := manifestData.NewPins[j]
-
 		return iPin.FileName < jPin.FileName
 	})
 
-	if format == "json" {
+	// Shorten the array for testing
+	if utils.IsTestMode() {
+		manifestData.NewPins = manifestData.NewPins[:100]
+	}
+
+	// TODO: if Root.to_file == true, write the output to a filename
+	// TODO: if Root.output == <fn>, write the output to a <fn>
+
+	if format == "" || format == "none" {
+		if utils.IsApiMode() {
+			format = "api"
+		} else {
+			format = "txt"
+		}
+	}
+
+	outFmt := "%s\t%s\t%s\n"
```

Comment:
Was there any problem with `Table`? 
Printing strings with tabs between them doesn't guarantee a tabular output and this is what happens in C++ (I have header and some rows misaligned with the rest). I thought that it was a bug and `Table` was a way to fix it. But right now I think that maybe we want TSV output and not a table? In such a case, let's remove Table and have CSV writer in `output` package. We can then configure this CSV writer to use tabs as delimeters (just like in `manifest` package)

Reply:
Yes. We want a file that can be imported into MS Excel.

But....now that I think about it....in the C++ code it can tell if it's being written to the console or not. If it is, it displays colored text, but if it's being re-directed (through `>`) or piped (through '|') we don't produce colored text. We could do the same thing here. I like the table view (with aligned columns), but it's not useful if redirected.

This is the line that knows if it's running against a terminal: https://github.com/TrueBlocks/trueblocks-core/blob/master/src/libs/utillib/options_base.cpp#L64.

Here's an article about it in golang: https://www.darkcoding.net/software/in-go-is-your-terminal-a-console-or-a-pipe-isatty-golang/

Reply:
Yeah, we could do that. But I think that we should at least keep format-specific logic in `output` package and use functions from there (to maintain consistency easily). For example, we could have `output.PrintTab`

Reply:
I think we may want to use an interface here in a way similar to the way `CBaseClass` works in the C++ code. So we can send into any `CArchive` and `CBaseClass` and it will handle it correctly. `CArchive` behaves exactly like a file in the C++ code. So, for example, I can do (with any data type):

```
cout << item << endl; // write to stdout
cout << item << endl; // write to stderr

ostringstream os;
os << item << endl; // to write to string

CArchive archive;
archive.Open(...)
archive << item << endl; // to write to file

archive >> different_item; // to read
```